api/proto/teleport/legacy/types/types.proto

```diff
@@ -5497,3 +5497,36 @@ message WatchStatusSpecV1 {
     (gogoproto.jsontag) = "kinds"
   ];
 }
+
+// ServerInfoV1 contains info that should be applied to joining Nodes.
+message ServerInfoV1 {
+  // Kind is the resource kind.
+  string Kind = 1 [(gogoproto.jsontag) = "kind"];
+  // SubKind is an optional resource subkind. Currently unused for this resource.
+  string SubKind = 2 [(gogoproto.jsontag) = "sub_kind,omitempty"];
+  // Version is the resource version.
+  string Version = 3 [(gogoproto.jsontag) = "version"];
+  // Metadata is the resource metadata.
+  Metadata Metadata = 4 [
+    (gogoproto.nullable) = false,
+    (gogoproto.jsontag) = "metadata"
+  ];
+  // Spec is the resource spec.
+  ServerInfoSpecV1 Spec = 5 [
+    (gogoproto.nullable) = false,
+    (gogoproto.jsontag) = "spec"
+  ];
+}
+
+// ServerInfoSpecV1 contains fields used to match Nodes to this ServerInfo.
+message ServerInfoSpecV1 {
+  // AWSInfo contains attributes to match to an EC2 instance.
+  message AWSInfo {
```

Comment:
nit: As this is within the context of `ServerInfoSpecV1`, I'd be tempted to just name this AWS for brevity. But this is a very minor nit.

Reply:
I'd prefer that too, but protobuf doesn't allow nested types and fields to have the same name.

Reply:
Oops, I forgot about that. I think I ran into similar on the join token stuff :(